adapters/rack/test/opentelemetry/adapters/rack/middlewares/tracer_middleware_test.rb

```diff
@@ -0,0 +1,186 @@
+      _(first_span.attributes['http.target']).must_equal '/'
+      _(first_span.status.canonical_code).must_equal OpenTelemetry::Trace::Status::OK
+      _(first_span.attributes['http.url']).must_equal 'http://example.org/'
+      _(first_span.name).must_equal '/'
+    end
+
+    it 'extracts parent context' do
+      _(first_span.parent_span_id).wont_equal '0000000000000000'
+    end
+
+    describe 'config[:allowed_request_headers]' do
+      let(:env) { Hash('HTTP_FOO_BAR' => 'http foo bar value') }
+
+      it 'defaults to nil' do
+        _(first_span.attributes['http.request.headers.foo_bar']).must_be_nil
+      end
+
+      describe 'when configured' do
+        let(:config) { default_config.merge(allowed_request_headers: ['foo_BAR']) }
+
+        it 'returns attribute' do
+          _(first_span.attributes['http.request.headers.foo_bar']).must_equal 'http foo bar value'
+        end
+      end
+    end
+
+    describe 'config[:allowed_response_headers]' do
+      let(:app) do
+        ->(_env) { [200, { 'Foo-Bar' => 'foo bar response header' }, ['OK']] }
+      end
+
+      it 'defaults to nil' do
+        _(first_span.attributes['http.response.headers.foo_bar']).must_be_nil
+      end
+
+      describe 'when configured' do
+        let(:config) { default_config.merge(allowed_response_headers: ['Foo-Bar']) }
+
+        it 'returns attribute' do
+          _(first_span.attributes['http.response.headers.foo_bar']).must_equal 'foo bar response header'
+        end
+
+        describe 'case-sensitively' do
+          let(:config) { default_config.merge(allowed_response_headers: ['fOO-bAR']) }
+
+          it 'returns attribute' do
+            _(first_span.attributes['http.response.headers.foo_bar']).must_equal 'foo bar response header'
+          end
+        end
+      end
+    end
+
+    describe 'config[:record_frontend_span]' do
+      describe 'default' do
+        it 'does not record span' do
+          _(exporter.finished_spans.size).must_equal 1
+        end
+      end
+
+      describe 'when recordable' do
+        let(:config) { default_config.merge(record_frontend_span: true) }
+        let(:env) { Hash('HTTP_X_REQUEST_START' => Time.now.to_i) }
+        let(:frontend_span) { exporter.finished_spans.last }
+
+        it 'records span' do
+          _(exporter.finished_spans.size).must_equal 2
+          _(frontend_span.name).must_equal 'http_server.queue'
+          _(frontend_span.attributes['service']).must_be_nil
+        end
+
+        describe 'when config[:web_service_name]' do
+          let(:config) do
+            default_config.merge(record_frontend_span: true,
+                                 web_service_name: 'my_service_name')
+          end
+
+          it 'records "service" attribute' do
+            _(frontend_span.attributes['service']).must_equal 'my_service_name'
+          end
+        end
+      end
+    end
+  end
+
+  describe 'config[:quantization]' do
```

Comment:
Do we still have quantization?

Reply:
We don't have a full Quantization implementation (reverted in 78acca7cd6d3274f7c0a352107c8413ad528e62f and a69f94ee834ccd6ca977b2e0abbed99af53d6bf0), but the callable `config[:quantization]` still exists, which is passed the `request_uri_or_path_info`.